### ISSUE TITLE
fix test issue: template folder location

### ISSUE THREAD

**Issue by GeorgGrassler** [itpplasma/NEO-2#5]
Currently the ctest and corresponding shellscript test_full_reconstruct.sh (/NEO-2/ShellScripts/test_full_reconstruct.sh) assume a ensemble of test templates as a basis to contruct the run folder for the tests. While for the PAR version these templates were relocated to /temp/AG-plasma/codes/neo-2_test_templates, some of the symbolic links for the QL version (Multispecies stuff) point to a (currently absend) Reference folder that should be in the same directory as neo-2_test_templates.

Way to fix:
* relocate the Reference (current location is probably the HOME directory of [nuclearfusionreactor] at ITP)
* fix references in test_full_reconstruct.sh to the template folders

**Comment by GeorgGrassler** [comment]
> some of the symbolic links for the QL version (Multispecies stuff) point to a (currently absend) Reference folder that should be in the same directory as neo-2_test_templates.
 
You mean /temp/AG-plasma/codes/Reference ?
You put it already there?

**Comment by krystophny** [comment]
*Automated triage pass (Claude), grounded in the repository at `80a6032`. Verify before acting.*

Still open, still real — but the fix proposed in 2023 (relocate a `Reference/` folder on the ITP `/temp` filesystem) is no longer the right fix. The template/reference breakage is now only one of five independent reasons `ShellScripts/test_full_reconstruct.sh` cannot run, and the test coverage it once provided has been replaced by the golden-record CI. Refining as: **remove the dead site-local ctest wiring**, and, if the QL multispecies reconstruct case is still wanted, move it into the golden-record suite.

## Purpose

`NEO-2-QL/CMakeLists.txt` and `NEO-2-PAR/CMakeLists.txt` still register 16 active `ADD_TEST` entries (plus one commented out) that shell out to `ShellScripts/test_full_reconstruct.sh`. Because the top-level `CMakeLists.txt` does `add_subdirectory(NEO-2-PAR)` / `add_subdirectory(NEO-2-QL)` (lines 116-117), these tests are part of the discovered CTest suite at the build root. Every one of them fails on any machine that is not an ITP workstation with a particular colleague's `/temp` tree intact — and fails on ITP machines too, for reasons unrelated to the missing `Reference/` folder.

Concretely, `ctest -N` at the build root lists 23 tests:

| Source | Count | Status |
|---|---|---|
| `TEST/CMakeLists.txt` (`boozmn_read_test`, `nrutil_test`, `er_rotation_test`, `spline_sparse_assembly_test`) | 4 | pass |
| `TEST/mympilib/CMakeLists.txt` (`test_mympilib`) | 1 | pass |
| `Test_Build_Code` (`NEO-2-QL/CMakeLists.txt:34`, `NEO-2-PAR/CMakeLists.txt:35`) | 2 | pass |
| `test_full_reconstruct.sh` entries (`NEO-2-QL:35-36`, `NEO-2-PAR:36-50`) | 16 | **permanently red** |

A CTest suite where 16 of 23 tests can never pass is worse than no suite: it trains everyone to ignore `ctest` output at the build root.

## Current state

The five independent breakages in `ShellScripts/test_full_reconstruct.sh`, all present at HEAD:

1. **Reference path** — the original complaint. `ShellScripts/test_full_reconstruct.sh:30-31`:
   ```bash
   referencepath='/temp/buchholz/Neo2/Testing/Reference/'
   #~ referencepath='../../../Reference/'
   ```
   The same hard-coded path is at `ShellScripts/test_scan.sh:23`.
2. **Executable name** — `ShellScripts/test_full_reconstruct.sh:33` sets `executablename="neo_2.x"` and line 143 does `cp ./$executablename ${testpath}/`. No such binary is produced by the current build system: `NEO-2-QL/CMakeLists.txt:7` sets `EXE_NAME "neo_2_ql.x"` and `NEO-2-PAR/CMakeLists.txt:7` sets `EXE_NAME "neo_2_par.x"`. Note that the *directory* is fine — no `RUNTIME_OUTPUT_DIRECTORY` is set anywhere, so the executables land in `build/NEO-2-QL/` and `build/NEO-2-PAR/`, which are exactly the default CTest working directories for these `ADD_TEST` calls. Only the file name is wrong.
3. **Template path** — `ShellScripts/test_full_reconstruct.sh:148`:
   ```bash
   cp -r -L /temp/AG-plasma/codes/neo-2_test_templates/${testcase}/* ./
   ```
   Introduced by `4b03c47` (2023-12-06, *"changed template location to /temp/AG-plasma/codes to be accessible for all users (still need to repair broken links)"*) — the commit that prompted this issue. `-L` dereferences the symlinks that the issue reports as dangling.
4. **Python import moved** — line 126 runs `from hdf5tools import compare_hdf5_files` in an inline `python3` heredoc. `hdf5tools.py` was moved into the installable package by `42f5263` (2024-09-18, *"shifted hdf5tools to proper neo2 pylibrary"*); it now lives at `python/src/neo2_util/hdf5tools.py` (`compare_hdf5_files` defined at line 1482) and is importable only as `neo2_util.hdf5tools`. `PythonScripts/hdf5tools.py` does not exist.
5. **Bare-name helper invocations** — line 74 calls `compare_data_files.py` and line 168 calls `switch_reconstruction.sh`, both without a path, i.e. they require `PythonScripts/` and `ShellScripts/` to be on `$PATH`. Nothing in the build or CI puts them there.

What replaced it: `.github/workflows/test-on-pr.yml` runs the golden-record suite (`make lorentz`, `make ql`, `make performance_ql`, `make par`) against test data cloned from `gitlab.tugraz.at/plasma/data.git` under `TESTS/NEO-2/golden_record/`, comparing HEAD against both the latest release tag and `main`. Tolerances as set in that workflow: `lorentz` uses `TOL=1e-14`, `LEGACY_TOL=1e-10`, `MAGNETICS_TOL=1e-12`; `ql` uses `TOL=1e-10`/`LEGACY_TOL=1e-8` against the release and `TOL=1e-15`/`LEGACY_TOL=1e-8` against `main`; `performance_ql` uses `TOL=1e-15`; `par` uses `TOL=1e-15`, `LEGACY_TOL=4e-8`. That is the same "run the code, compare HDF5 output to a reference within a tolerance" idea, done with data that is versioned and reachable from CI.

The in-repo unit suite is `TEST/CMakeLists.txt` plus `TEST/mympilib/CMakeLists.txt`. `Makefile:27` defines `test:` as `cd build && ctest --test-dir TEST --output-on-failure`, and `.github/workflows/unit-tests.yml:80` runs exactly `make test` — so CI deliberately never reaches the QL/PAR entries. They are red only for whoever runs `ctest` at the build root.

## Reproduction / evidence

```bash
git clone https://github.com/itpplasma/NEO-2.git && cd NEO-2
make                                  # builds build/NEO-2-QL/neo_2_ql.x, build/NEO-2-PAR/neo_2_par.x
cd build && ctest -N                  # lists 23 tests, 16 of them shell-script entries
ctest -R Extended_QL_Multispecies --output-on-failure
```

Observed on any machine without `/temp/$LOGNAME/Neo2/Testing/Runs/` and without `NEO2_TEST_PATH` set, from `ShellScripts/test_full_reconstruct.sh:11-14`:

```
There is no directory defined for storing test data. Please create the
/temp/$LOGNAME/Neo2/Testing/Runs/ directory or define a new location in
the environment variable NEO2_TEST_PATH.
```
exit code 1.

With `NEO2_TEST_PATH` pointed at an existing directory the script advances past the guard and dies at line 143 with `cp: cannot stat './neo_2.x'`. Neither failure has anything to do with the missing `Reference/` folder — the script never gets far enough to consult `referencepath`.

Expected after this issue: `ctest -N` at the build root lists only tests that can pass on a clean checkout, and `ctest --output-on-failure` at the build root is green.

## Behaviour (BDD)

```gherkin
Scenario: Clean checkout has a green CTest suite at the build root
  Given a fresh clone of NEO-2 at main on a machine with no /temp/AG-plasma tree
    And NEO2_TEST_PATH is unset
  When the developer runs "make" and then "cd build && ctest --output-on-failure"
  Then ctest reports 100% tests passed
    And "ctest -N" lists exactly 7 tests: boozmn_read_test, nrutil_test,
        er_rotation_test, spline_sparse_assembly_test, test_mympilib and the
        two Test_Build_Code entries
    And no test named Extended_QL_Multispecies, Extended_QL_Multispecies_Read,
        Test_Testcase1-Seq, Test_Testcase1-Par, Test_Relativistic_BeliaevBudker-Seq,
        Test_Relativistic_BeliaevBudker-Par, Test_Relativistic_BraamsKarney-Seq,
        Test_Relativistic_BraamsKarney-Par, or Test_par_{standard_polynomials,
        quadratic_polynomials,cubic_splines,general_bsplines}-{Seq,Par} appears

Scenario: make test is unaffected
  Given the same clean checkout
  When the developer runs "make test"
  Then ctest runs exactly boozmn_read_test, nrutil_test, er_rotation_test,
       spline_sparse_assembly_test and test_mympilib
    And all of them pass

Scenario: No dangling references to the removed harness remain
  Given the tree after the change
  When "rg -n 'test_full_reconstruct|test_scan' --glob '!.git' ." is run at the repo root
  Then it produces zero hits

Scenario: Golden-record CI still covers the QL path
  Given a pull request against main
  When the "Run Test" workflow in .github/workflows/test-on-pr.yml executes
       "make ql" in TESTS/NEO-2/golden_record/
  Then the comparison against the latest release passes with TOL=1e-10 and
       LEGACY_TOL=1e-8
    And the comparison against main passes with TOL=1e-15 and LEGACY_TOL=1e-8
```

## Implementation plan

1. **`NEO-2-QL/CMakeLists.txt`** — delete lines 35, 36 and 37 (`Extended_QL_Multispecies`, `Extended_QL_Multispecies_Read`, and the commented-out `test_scan.sh` line). **Leave line 34 (`Test_Build_Code`) untouched** — it belongs to itpplasma/NEO-2#137.
2. **`NEO-2-PAR/CMakeLists.txt`** — delete lines 36 through 50 inclusive (the six `Test_Testcase1-*` / `Test_Relativistic_*` entries, the blank line 42, and the eight `Test_par_*` entries). **Leave line 35 (`Test_Build_Code`) untouched** — also #137.
3. **Delete `ShellScripts/test_full_reconstruct.sh` and `ShellScripts/test_scan.sh`.** After steps 1-2 they have no callers: at HEAD, `grep -rn 'test_full_reconstruct|test_scan' --exclude-dir=.git .` hits only those two `CMakeLists.txt` files, and nothing else in the tree references either script.
4. **Delete `ShellScripts/create_reference.sh` as well.** It exists solely to generate the md5sum reference trees that the deleted harness consumed, and it is itself already dead: `create_reference.sh:8` sets `executable=neo_2.x`, the same binary that no longer exists, and no file in the repository invokes it. If maintainers prefer to keep it as a manual-use snippet, leave it and say so in the commit message — this is the one judgement call in the PR.
5. **Keep `ShellScripts/switch_reconstruction.sh`.** It is still called by `ShellScripts/create_reference.sh:20` and documented at `DOC/ExtraDocuments/neo-2-par-eccd_runs.tex:656`. If step 4 deletes `create_reference.sh`, the `.tex` reference is the only remaining one — still a reason to keep the script.
6. **Keep `PythonScripts/compare_data_files.py`.** After step 3 its only in-repo caller is gone, but removing it is out of scope here; leave it for a separate cleanup.
7. **Documentation** — no change is required for this PR. `compare_data_files` appears nowhere in `DOC/`. `hdf5tools` is mentioned in `DOC/UserDoc/Running.tex:179`, `DOC/ExtraDocuments/neo-2-ql-demo_runs.tex` (lines 174, 202, 213, 744, 747, 852-853, 870) and `DOC/ExtraDocuments/neo-2-par-eccd_runs.tex` (lines 186, 214, 223, 655, 688-689), but all of those describe the Python module for users, not the ctest harness. Updating those `import hdf5tools` snippets to `from neo2_util import hdf5tools` is a genuine but separate doc bug — open it as its own issue rather than mixing it in here.
8. **Optional, only if the multispecies reconstruct case is genuinely still wanted as coverage** — do *not* resurrect the shell harness. Instead add a `multispec_reconstruct` target to the golden-record `Makefile` in `gitlab.tugraz.at/plasma/data.git` under `TESTS/NEO-2/golden_record/`, mirroring the existing `ql` target, with the run template committed as LFS data there; then add a step to `.github/workflows/test-on-pr.yml` after the existing `make ql` step, guarded by `if: github.event_name == 'pull_request'` like the `par` step. This is a **data-repo PR first**, NEO-2 PR second — the workflow step will fail until the data-repo target exists.
9. No new test file is needed: the acceptance criterion is that the existing suite becomes green and `ctest -N` shrinks from 23 to 7.

If maintainers prefer preservation over deletion, the minimum viable alternative is to keep the two scripts but (a) gate every `ADD_TEST` behind an `option(NEO2_SITE_TESTS "Enable ITP site-local reconstruct tests" OFF)`, (b) replace `test_full_reconstruct.sh` lines 30, 33 and 148 with `${NEO2_REFERENCE_PATH}`, `${NEO2_EXECUTABLE}` and `${NEO2_TEMPLATE_PATH}` read from the environment with no defaults, (c) fix line 126 to `from neo2_util.hdf5tools import compare_hdf5_files`, and (d) give lines 74 and 168 explicit paths. That is strictly more work for coverage the golden record already provides, so deletion is recommended.

## Acceptance criteria

- [ ] `grep -rn 'test_full_reconstruct\|test_scan' --exclude-dir=.git .` at the repo root returns zero hits.
- [ ] `cd build && ctest -N` lists exactly 7 tests: `boozmn_read_test`, `nrutil_test`, `er_rotation_test`, `spline_sparse_assembly_test`, `test_mympilib`, and two `Test_Build_Code`.
- [ ] `cd build && ctest --output-on-failure` exits 0 on a clean Ubuntu 24.04 runner with no `/temp` tree and `NEO2_TEST_PATH` unset.
- [ ] `make test` still runs and passes all five unit tests.
- [ ] `NEO-2-QL/CMakeLists.txt:34` and `NEO-2-PAR/CMakeLists.txt:35` (`Test_Build_Code`) are byte-identical to HEAD — this PR must not touch them.
- [ ] `.github/workflows/unit-tests.yml` and `.github/workflows/test-on-pr.yml` both pass; golden-record `lorentz`, `ql`, `par`, `performance_ql` show no numerical change (tolerances unchanged).
- [ ] `git diff --stat` shows no `*.f90` / `*.F90` file touched — only `CMakeLists.txt` edits and `ShellScripts/*.sh` deletions.

## References

No specific literature reference; this is test-infrastructure cleanup internal to the code. The replacement mechanism is already in the repository — see `.github/workflows/test-on-pr.yml` (golden-record comparison against `gitlab.tugraz.at/plasma/data.git`) and `.github/workflows/unit-tests.yml` (`make test`).

## Scope and risk

Single small PR, deletions only, no Fortran touched — the numerical results cannot change. The only risk is deleting a script someone still runs by hand on an ITP workstation; steps 5 and 6 keep `switch_reconstruction.sh` and `compare_data_files.py` precisely to bound that, and step 4 flags `create_reference.sh` as the one item worth asking about before deleting.

Merge-order note: itpplasma/NEO-2#137 is reported to edit the adjacent `Test_Build_Code` lines in these same two files. That cross-reference could not be verified from the checkout alone — confirm on GitHub before sequencing. Steps 1 and 2 are written to leave those two lines alone so that any conflict is trivial.
